packages/block-editor/src/components/block-navigation/use-block-navigation-client-ids.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * WordPress dependencies
+ */
+
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { isClientIdSelected } from './utils';
+import { store as blockEditorStore } from '../../store';
+
+const useBlockNavigationSelectedClientIds = (
+	selectedBlockClientIds,
+	__experimentalPersistentListViewFeatures
+) =>
+	useSelect(
+		( select ) => {
+			const {
+				getSelectedBlockClientId,
+				getSelectedBlockClientIds,
+			} = select( blockEditorStore );
+
+			if ( selectedBlockClientIds ) {
+				return selectedBlockClientIds;
+			}
+
+			if ( __experimentalPersistentListViewFeatures ) {
```

Comment:
Does this make any difference somewhere? `getSelectedBlockClientIds` will first check if it's single block selected and return it in an Array.

Reply:
This is a confusing function 😬 I'd expect it to return an `Array` in all cases. Yet we return a single string on line 32.

We should just use `getSelectedBlockClientIds()` directly. We can also remove all the `Array.isArray` checks and just check for `length` property.

Reply:
Well what do you know, I think you're right! 😄

Reply:
Or actually...

The problem here is that some consumers (Dropdown to be specific) really expect the selected client ID to be **single and string, or null otherwise**, whereas the Persistent List View wants the whole multiselection array.

For example, `getBlockHierarchyRootClientId` (used [here](https://github.com/WordPress/gutenberg/pull/31290/files/721c63202ebe2c8e38a5d54594c32445c6150ab8#diff-571fdc70e9d5e56e4917e66740dbf811ad28bf4b905e02d331f9b191fa49146cR61) and defined [here](https://github.com/WordPress/gutenberg/blob/b65583def3168fb481b2a776f892f2a38b10a616/packages/block-editor/src/store/selectors.js#L589-L605)) will use the parameter straight away as object property.

Using those two distinct functions basically saves a bunch of additional checks and conversions to make sure everything keeps receiving the parameters with the expected types.

This said, I think the major cause of confusion here is this:
```js
const isSingleBlockSelected =
	selectedClientIds && ! Array.isArray( selectedClientIds );
```
Single selections can be arrays as well, and the naming here doesn't make it clear.
I guess `isNotMultiselectedBlock` would not make it much clearer, so I'm more oriented to just adding a comment to add some context.